wp-includes/l10n.php

```diff
@@ -694,6 +707,63 @@ function translate_nooped_plural( $nooped_plural, $count, $domain = 'default' )
 	}
 }
 
+/**
+ * Load a .mo file from file previously processed by 
+ * the {@link https://developer.wordpress.org/reference/functions/load_textdomain/
+ * load_textdomain() function}.
+ * 
+ * @param string $domain Text domain. Unique identifier for retrieving translated strings.
+ * @return bool True on success, false on failure.
+ */
+function load_textdomain_from_cache( $domain ) {
+	return get_transient( WP_l10n_CACHE_PREFIX . $domain );
+}
+
+/**
+ * Cache a .mo file via transient API as previously processed by 
+ * the {@link https://developer.wordpress.org/reference/functions/load_textdomain/
+ * load_textdomain() function}.
+ * 
+ * @param string $domain Text domain. Unique identifier for retrieving translated strings.
+ * @param string $moobject A MO object
+ * @return bool True on success, false on failure.
+ */
+function cache_textdomain( $domain, $moobject ) {
+	
+	$option_name = WP_l10n_CACHE_PREFIX . 'domains_cached';
+	$key = WP_l10n_CACHE_PREFIX . $domain;
+	$keys = get_option( $option_name, [] );
```

Comment:
That's the place I am concerned about. I don't feel the option storage is the correct place to keep the list of text domain to use for cache clearing. In the current approach the option storage can keep domains which are already removed and try to clear the cache for them. One of the solutions is hooking into a "domain removed" hook (if exists), and then reduce the list.